src/yaml/from_yaml.cr

```diff
@@ -21,16 +21,23 @@ def Array.from_yaml(string_or_io)
 end
 
 def Nil.new(pull : YAML::PullParser)
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_scalar
+  when "", "NULL", "Null", "null", "~"
     nil
   else
     raise YAML::ParseException.new("Expected nil, not #{value}", 0, 0)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  case value = pull.read_scalar
+  when "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"
```

Comment:
[10.3.2](http://yaml.org/spec/1.2/spec.html#id2805071) states that only "true", "True" and "TRUE" represent the boolean true value.